apps/src/JSInterpreter.js

```diff
@@ -372,13 +372,17 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
     if ((reachedBreak && !unwindingAfterStep) ||
         (doneUserLine && !unwindingAfterStep && !atMaxSpeed) ||
         this.yieldExecution ||
+        this.interpreter.paused_ ||
```

Comment:
HELL TO THE YES SYNC FUNCTIONS

(I'm kind of excited about this)

On Wed, Feb 10, 2016 at 3:12 PM, Josh Lory <notifications@github.com> wrote:

> In apps/src/JSInterpreter.js
> <https://github.com/code-dot-org/code-dot-org/pull/6746#discussion_r52542362>
> :
>
> > @@ -372,13 +372,17 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
> >      if ((reachedBreak && !unwindingAfterStep) ||
> >          (doneUserLine && !unwindingAfterStep && !atMaxSpeed) ||
> >          this.yieldExecution ||
> > +        this.interpreter.paused_ ||
>
> Can this.yieldExecution be removed, and JSInterpreter.prototype.yield
> implemented with this.interpreter.paused_?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/6746/files#r52542362>.
>



-- 
Mehal Shah